configs.go

```diff
@@ -182,64 +167,72 @@ func obfuscateConfig() {
 	if err := obfuscateData(&conf.Password); err != nil {
 		fail(err.Error())
 	}
-	for i, check := range conf.Check {
-		if err := obfuscateData(&conf.Check[i].Message); err != nil {
+	for i := range conf.Checks {
+		if err := obfuscateData(&conf.Checks[i].Message); err != nil {
 			fail(err.Error())
 		}
-		if conf.Check[i].Hint != "" {
-			if err := obfuscateData(&conf.Check[i].Hint); err != nil {
-				fail(err.Error())
-			}
-		}
-		for j := range check.Pass {
-			if err := obfuscateCond(&conf.Check[i].Pass[j]); err != nil {
+		if conf.Checks[i].Hint != "" {
+			if err := obfuscateData(&conf.Checks[i].Hint); err != nil {
 				fail(err.Error())
 			}
 		}
-		for j := range check.PassOverride {
-			if err := obfuscateCond(&conf.Check[i].PassOverride[j]); err != nil {
-				fail(err.Error())
-			}
-		}
-		for j := range check.Fail {
-			if err := obfuscateCond(&conf.Check[i].Fail[j]); err != nil {
-				fail(err.Error())
-			}
+		if err := obfuscateCond(conf.Checks[i].Condition); err != nil {
+			fail(err.Error())
 		}
 	}
 }
 
 // obfuscateCond is a convenience function to obfuscate all string fields of a
 // struct using reflection. It assumes all struct fields are strings.
-func obfuscateCond(c *cond) error {
+
+// ummmmmm
+func obfuscateCond(c aeaconf2.Condition) error {
 	s := reflect.ValueOf(c).Elem()
+	t := s.Type()
+
 	for i := 0; i < s.NumField(); i++ {
-		if s.Type().Field(i).Name == "regex" {
-			continue
-		}
-		datum := s.Field(i).String()
-		if err := obfuscateData(&datum); err != nil {
-			return err
+		field := s.Field(i)
+		fieldType := t.Field(i).Type
+
+		if fieldType.Kind() == reflect.String {
+			datum := field.String()
+			if err := obfuscateData(&datum); err != nil {
+				return err
+			}
+			field.SetString(datum)
+		} else if fieldType.Implements(reflect.TypeOf((*aeaconf2.Condition)(nil)).Elem()) {
+			if err := obfuscateCond(field.Addr().Interface().(aeaconf2.Condition)); err != nil {
+				return err
+			}
 		}
-		s.Field(i).SetString(datum)
 	}
+
 	return nil
 }
 
 // deobfuscateCond is a convenience function to deobfuscate all string fields
 // of a struct using reflection.
-func deobfuscateCond(c *cond) error {
+func deobfuscateCond(c aeaconf2.Condition) error {
 	s := reflect.ValueOf(c).Elem()
+	t := s.Type()
+
 	for i := 0; i < s.NumField(); i++ {
-		if s.Type().Field(i).Name == "regex" {
-			continue
-		}
-		datum := s.Field(i).String()
-		if err := deobfuscateData(&datum); err != nil {
-			return err
+		field := s.Field(i)
+		fieldType := t.Field(i).Type
+
+		if fieldType.Kind() == reflect.String {
+			datum := field.String()
+			if err := deobfuscateData(&datum); err != nil {
+				return err
+			}
+			field.SetString(datum)
+		} else if fieldType.Implements(reflect.TypeOf((*aeaconf2.Condition)(nil)).Elem()) {
```

Comment:
This may look better as a function on the aeaconf struct. Maybe something like `ModifyFields` that takes a `func(string) string` as a parameter? Then we don't have to do the Implements stuff

Reply:
Yeah I could probably just do that on the `aeaconf` side

Reply:
Added on `v0.1.2` (`compat.ModifyConditionStrings`)